go/worker/registration/worker.go

```diff
@@ -1128,15 +1112,9 @@ func New(
 		registerCh:         make(chan struct{}, 64),
 	}
 
-	if viper.GetBool(CfgRegistrationForceRegister) {
-		if err = SetForcedDeregister(w.store, false); err != nil {
-			return nil, fmt.Errorf("failed to clear persisted forced-deregister: %w", err)
-		}
-		storedDeregister = false
-	}
 	w.storedDeregister = storedDeregister
 
-	if flags.ConsensusValidator() {
+	if config.GlobalConfig.Consensus.Validator {
```

Comment:
This should also work if you set the global mode to validator?

Reply:
The previous code only checked the flag:
https://github.com/oasisprotocol/oasis-core/blob/master/go/oasis-node/cmd/common/flags/flags.go#L75-L77

Reply:
Yeah this was wrong, we should make sure that if you set the global mode to validator your node will actually become a validator :-)